src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1055,7 +1055,7 @@ protected function sibling(\DOMNode $node, string $siblingDir = 'nextSibling'):
 
     private function parseHtml5(string $htmlContent, string $charset = 'UTF-8'): \DOMDocument
     {
-        return $this->html5Parser->parse($this->convertToHtmlEntities($htmlContent, $charset), [], $charset);
+        return $this->html5Parser->parse($this->convertToHtmlEntities($htmlContent, $charset), ['encoding' => $charset]);
```

Comment:
Is this a bugfix? Should we fix that on a lower branch then?

Reply:
Good point, I did it here without thinking too much but it makes sense to treat it as a bugfix

Reply:
@tgalopin Can you create another PR for this bug fix?

Reply:
Yes, that was my plan :)

Reply:
Done: https://github.com/symfony/symfony/pull/44180